### ISSUE TITLE
Mupx Setup Failure - Error starting userland proxy: listen tcp 127.0.0.1:27017

### ISSUE THREAD

**Issue by chackerian** [arunoda/meteor-up-legacy#596]
Receiving the following error when I run mupx setup. I am at the latest version of mupx and meteor and my server credentials seem to be correct as this command creates folders within /opt although no files are created within the folders.

<img width="639" alt="screen shot 2015-08-12 at 12 55 35 pm" src="https://cloud.githubusercontent.com/assets/5833874/9230701/d99522fa-40f1-11e5-93de-d3cc9d7a68c5.png">

**Comment by MasterJames** [comment]
If you are changing from mup to mupx you need to follow the migration guide.
If you are running g your own mongo set in setting gs to false and then include mongo_url setting as well. 
You may also try to stop mupx and or mongo manually before redeploying I think.

**Comment by chackerian** [comment]
I followed the migration guide and I am not currently using a foreign mongo db. I'm just using the local one included in meteor.

**Comment by MasterJames** [comment]
Maybe post your settings then.
mupx stop
DEBUG=\* mupx setup
DEBUG=\* mupx deploy

and then POST logs from that.

**Comment by chackerian** [comment]
```
-----------------------------------STDERR-----------------------------------
Error response from daemon: no such id: spacetrades
Error: failed to remove containers: [spacetrades]
Error response from daemon: no such id: spacetrades-frontend
Error: failed to remove containers: [spacetrades-frontend]
Error response from daemon: Cannot start container d0571d13fc68a555a00e75a85c83cc901f54dc79a215c6f0dc01027d008eada7: Error starting userland proxy: listen tcp 0.0.0.0:80: bind: address already in use
-----------------------------------STDOUT-----------------------------------
base: Pulling from meteorhacks/meteord
4c8cbfd2973e: Already exists
60c52dbe9d91: Already exists
66b378b5905c: Already exists
715693557c06: Already exists
16e0fe9fbd39: Already exists
cd378c1c4783: Already exists
041bc56089e7: Already exists
b545df9bb341: Already exists
b545df9bb341: Already exists
Digest: sha256:44960a933248c70b2e6a69df8fac4d3f0c521299ef153186552f46d27053385c
Status: Image is up to date for meteorhacks/meteord:base
d0571d13fc68a555a00e75a85c83cc901f54dc79a215c6f0dc01027d008eada7
----------------------------------------------------------------------------
```

**Comment by MasterJames** [comment]
Right still need settings. Seems you probably want Mongo = false
Since it's running from elsewhere already.

**Comment by MasterJames** [comment]
Try this command from ssh main folder
meteor mongo
See 
help
db.help ()
show logs
show log global

Please be certain you're using mupx and not mup and you'very follow the migration instructions.
If it's a newl'y deployed instance and you still have that problem it's very strange.
At that point you'll need to check what version of Ubuntu you're deploying to and it's not corrupted.

**Comment by chackerian** [comment]
Looks like mongod actually isn't running ( I did ps aux | grep mongo and didn't see anything on the server). meteor mongo from the main folder does nothing. Could I manually start mongo on the server and if so, what is the --dbpath folder that mupx uses.

**Comment by MasterJames** [comment]
Try meteorpedia.com/read/Mongo
PROJECT_DIR/.meteor/local/dB
It runs locally with command "meteor", but is not running after deployment? 
Please post your mupx settings.
Particularly does it contain 
setupMongo = true

**Comment by chackerian** [comment]
Here is the mup.json file and yes setupMongo is set to true. I tried changing it to false and running setup and deploying but it failed with the same error message as before. Also, PROJECT_DIR/.meteor/local/dB is the directory path locally but would you know where it is stored after deploy on the server ( /opt/spacetrades/ doesn't have a .meteor folder). 

```
"servers": [
    {
      "host": "54.187.98.241",
      "username": "ubuntu",
      // "password": "password",
      // or pem file (ssh based authentication)
      // WARNING: Keys protected by a passphrase are not supported
      "pem": "~/Desktop/AWS/clavis.pem",
      // Also, for non-standard ssh port use this
      //"sshOptions": { "port" : 49154 },
      // server specific environment variables
      "env": {}
    }
  ],

  // Install MongoDB on the server. Does not destroy the local MongoDB on future setups
  "setupMongo": true,

  // Application name (no spaces).
  "appName": "spacetrades",

  // Location of app (local directory). This can reference '~' as the users home directory.
  // i.e., "app": "~/Meteor/my-app",
  // This is the same as the line below.
  "app": ".",

  // Configure environment
  // ROOT_URL must be set to your correct domain (https or http)
  "env": {
    "PORT": 80,
     "ROOT_URL": "http://ec2-52-24-238-181.us-west-2.compute.amazonaws.com/"
     // "MONGO_URL": "mongodb://localhost:27017/spacetrades"
  },

  // Meteor Up checks if the app comes online just after the deployment.
  // Before mup checks that, it will wait for the number of seconds configured below.
  "deployCheckWaitTime": 15,

  // show a progress bar while uploading. 
  // Make it false when you deploy using a CI box.
  "enableUploadProgressBar": true
}
```

**Comment by MasterJames** [comment]
If you put false I believe you also need the MONGO_URL

**Comment by chackerian** [comment]
I tried it with that line un-commented with the URL and it persisted to throw errors. Turns out my web app had an error in it that was likely crashing the entire site although when tested locally it ran fine. Additionally, I ran the start.sh file manually from the server. One of these two things seems to have solved the issue and deploys are executing just fine. Thanks for responding quickly to this issue and helping me troubleshoot.
